Sources/TuistKit/Utils/Functions.swift

```diff
@@ -0,0 +1,36 @@
+/// Produces a getter function for a given key path. Useful for composing property access with functions.
```

Comment:
What about moving this extension to the [`Extensions` directory](https://github.com/tuist/tuist/tree/master/Sources/TuistCore/Extensions) of `TuistCore`?